src/main/java/de/tum/in/www1/artemis/service/connectors/localci/buildagent/BuildJobExecutionService.java

```diff
@@ -453,21 +456,45 @@ private BuildResult constructBuildResult(List<BuildResult.LocalCITestJobDTO> fai
     }
 
     private Path cloneRepository(VcsRepositoryUri repositoryUri, @Nullable String commitHash, boolean checkout, String buildJobId) {
+        int attempt = 0;
+        Repository repository = null;
+
+        // Ignore warning about always true condition, as the loop will break if the clone is successful
+        while (attempt < MAX_CLONE_RETRIES) {
+            try {
+                attempt++;
+                // Clone the assignment repository into a temporary directory
+                // TODO: use a random value if commitHash is null
+                repository = buildJobGitService.cloneRepository(repositoryUri, Path.of(CHECKED_OUT_REPOS_TEMP_DIR, commitHash, repositoryUri.folderNameForRepositoryUri()));
+
+                // Break out of the loop if clone is successful
+                break;
+            }
+            catch (GitAPIException | IOException | URISyntaxException e) {
+                if (attempt >= MAX_CLONE_RETRIES) {
+                    String msg = "Error while cloning repository " + repositoryUri.repositorySlug() + " with uri " + repositoryUri + " after " + MAX_CLONE_RETRIES + " attempts";
+                    buildLogsMap.appendBuildLogEntry(buildJobId, msg);
+                    throw new LocalCIException(msg, e);
+                }
+                // Log the retry attempt
```

Comment:
I think this comment is fairly redundant, as a log statement is self-explanatory.